pkg/operator2/configobservation/console/observe_consoleurl.go

```diff
@@ -0,0 +1,43 @@
+package console
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+func ObserveConsoleURL(genericlisters configobserver.Listers, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, []error) {
+	assetPublicURLPath := []string{configobservation.OAuthServerConfigPrefix, "oauthConfig", "assetPublicURL"}
+	listers := genericlisters.(configobservation.Listers)
+	errs := []error{}
+
+	currentAssetURL, _, err := unstructured.NestedString(previouslyObservedConfig, assetPublicURLPath...)
+	if err != nil {
+		return previouslyObservedConfig, append(errs, err)
```

Comment:
we might want to continue on an error  and try to fix potentially broken `previouslyObservedConfig`